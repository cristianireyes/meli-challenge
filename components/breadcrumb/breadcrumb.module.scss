.breadcrumb {
  display: flex;
  color: #666;
  font-size: 14px;
}

.categories {
  display: flex;
  padding-left: 0rem;
  flex-wrap: wrap;
  line-height: 1.25rem;
}

.category {
  margin-right: 0.5rem;

  &:after { content: " >"; }
  &:last-child:after { content: ''; }
}