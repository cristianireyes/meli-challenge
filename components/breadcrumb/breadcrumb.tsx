import { FunctionComponent } from 'react';
import { nanoid } from 'nanoid';
import styles from './breadcrumb.module.scss';

interface BreadcrumbProps {
  categories: string[];
}

export const Breadcrumb: FunctionComponent<BreadcrumbProps> = ({
  categories,
}) => (
  <div className={styles.breadcrumb}>
    <ul className={styles.categories}>
      {categories.map(category => {
        return (
          <li className={styles.category} key={`${nanoid()}-${category}`}>
            {category}
          </li>
        );
      })}
    </ul>
  </div>
);
