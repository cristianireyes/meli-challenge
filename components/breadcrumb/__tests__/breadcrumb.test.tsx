import { render } from '@testing-library/react';
import { Breadcrumb } from '../breadcrumb';

/**
 * Mocks
 */
jest.mock('nanoid', () => {
  return { nanoid: () => '1234' };
});
const categories = ['Deportes y Fitness', 'Ciclismo', 'Bicicletas'];

/**
 * Tests
 */

describe('breadcrumb component', () => {
  it('should display a list of categories', () => {
    const { container, getByText } = render(
      <Breadcrumb categories={categories} />,
    );

    expect(getByText(categories[0])).toBeDefined();
    expect(getByText(categories[1])).toBeDefined();
    expect(getByText(categories[2])).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
