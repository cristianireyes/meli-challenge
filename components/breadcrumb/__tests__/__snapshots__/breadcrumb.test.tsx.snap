// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumb component should display a list of categories 1`] = `
<div>
  <div
    class="breadcrumb"
  >
    <ul
      class="categories"
    >
      <li
        class="category"
      >
        Deportes y Fitness
      </li>
      <li
        class="category"
      >
        Ciclismo
      </li>
      <li
        class="category"
      >
        Bicicletas
      </li>
    </ul>
  </div>
</div>
`;
